driver-sync/src/test/functional/com/mongodb/client/AbstractClientSideEncryptionKmsTlsTest.java

```diff
@@ -116,5 +118,103 @@ public void testInvalidKmsCertificate() {
             assertTrue(expectedKmsTlsError.causeContainsExpectedMessage(e));
         }
     }
+
+    @Test()
+    public void testThatCustomSslContextIsUsed() {
```

Comment:
These tests don't exactly match the recommended prose tests, but they are simpler and I think achieve the desired effect: to prove the the configured SSLContext is actually used.